src/transform/src/equivalence_propagation.rs

```diff
@@ -68,6 +70,15 @@ impl crate::Transform for EquivalencePropagation {
             &mut get_equivalences,
         );
 
+        if prior == *relation {
+            let ck = crate::ColumnKnowledge::default();
+            ck.transform(relation, ctx)?;
+            mz_ore::soft_assert_or_log!(
+                prior == *relation,
+                "ColumnKnowledge performed work after EquivalencePropagation"
```

Comment:
I think this would be better to do after the `mz_repr::explain::trace_plan(&*relation);`, so that we can observe EQProp and CK separately in the optimizer trace tool.

Reply:
Ah, I don't understand these tools, but I am totally flexible here. I'm optimistic that we delete the code in a week, though.